File: src/bmsx_cpp/machine/runtime/cpu_state.cpp
#include "machine/runtime/cpu_state.h"

#include "machine/runtime/runtime.h"

namespace bmsx {

// disable-next-line single_line_method_pattern -- runtime save-state API keeps CPU/module-cache coupling out of callers.
CpuRuntimeState captureRuntimeCpuState(const Runtime& runtime) {
	return runtime.machine.cpu.captureRuntimeState(runtime.m_moduleCache);
}

// disable-next-line single_line_method_pattern -- runtime save-state API keeps CPU/module-cache coupling out of callers.
void applyRuntimeCpuState(Runtime& runtime, const CpuRuntimeState& state) {
	runtime.machine.cpu.restoreRuntimeState(state, runtime.m_moduleCache);
}

} // namespace bmsx

File: packages/bmsx-console/src/machine/runtime/cpu_state.ts
import type { CpuRuntimeState } from '../cpu/cpu';
import type { Runtime } from './runtime';

// disable-next-line single_line_method_pattern -- runtime save-state API keeps CPU/module-cache coupling out of callers.
export function captureRuntimeCpuState(runtime: Runtime): CpuRuntimeState {
	return runtime.machine.cpu.captureRuntimeState(runtime.moduleCache);
}

// disable-next-line single_line_method_pattern -- runtime save-state API keeps CPU/module-cache coupling out of callers.
export function applyRuntimeCpuState(runtime: Runtime, state: CpuRuntimeState): void {
	runtime.machine.cpu.restoreRuntimeState(state, runtime.moduleCache);
}
